lightning/src/chain/keysinterface.rs

```diff
@@ -1043,12 +1036,25 @@ impl KeysManager {
 					}
 					spend_tx.input[input_idx].witness = keys_cache.as_ref().unwrap().0.sign_dynamic_p2wsh_input(&spend_tx, input_idx, &descriptor, &secp_ctx)?;
 				},
-				SpendableOutputDescriptor::StaticOutput { ref output, .. } => {
-					let derivation_idx = if output.script_pubkey == self.destination_script {
```

Comment:
We still need to support spending the old-style derivation, though. So now we'll have three possible scripts to match against instead of two.

Reply:
I'm kinda confused about this, I thought we were gonna bump up the version so that older versions won't be affected by this change. Does this mean I'll have the destination script created for the `KeysManager::new()` and create a fresh one here in `spend_spendable_outputs`?

Reply:
Hmm, I suppose we could always create a `KeysManagerV2` and only make this change for that new struct, but if we can avoid it it'd be nice to, IMO. And I don't think its a *ton* more work to be backwards-compatible here.

Reply:
Would this suffice?

Reply:
No, I don't think so sadly, users with newer software will see a `channel_keys_id`, but may have opened a channel with older software, causing us to not do the old-style lookup.

Reply:
Oh, I did not account for that. How do we figure out if the channel was opened on the older version? 🤔

Reply:
Just check whether the script being spent matches the old scripts or if they match the new scripts :)

Reply:
Great that makes sense, thank you for clearing it up!

Reply:
Okay, so I have just added an extra condition where we check the spend script with the KeyManager's destination script as well. So rn it's something like this:

1.  If channel_keys_id is not provided, destination_script defaults to the old one. The spent output is checked against the destination script and the derivation_idx is assigned to either 1 or 2.
2. If the channel_keys_id is provided, destination_script is made fresh, the spent output is checked against both old and new (in case the channel is on older software), and if one of them matches derivation_idx is 1 or else it assigns 2.